examples/Python3/snowboydecoder.py

```diff
@@ -163,20 +175,63 @@ def audio_callback(in_data, frame_count, time_info, status):
                 time.sleep(sleep_time)
                 continue
 
-            ans = self.detector.RunDetection(data)
-            if ans == -1:
+            status = self.detector.RunDetection(data)
+            if status == -1:
                 logger.warning("Error initializing streams or reading audio data")
-            elif ans > 0:
-                message = "Keyword " + str(ans) + " detected at time: "
-                message += time.strftime("%Y-%m-%d %H:%M:%S",
+
+            #small state machine to handle recording of phrase after keyword
+            if state == "PASSIVE":
+                if status > 0: #key word found
+                    self.recordedData = []
+                    self.recordedData.append(data)
+                    silentCount = 0
+                    message = "Keyword " + str(status) + " detected at time: "
+                    message += time.strftime("%Y-%m-%d %H:%M:%S",
                                          time.localtime(time.time()))
-                logger.info(message)
-                callback = detected_callback[ans - 1]
-                if callback is not None:
-                    callback()
+                    logger.info(message)
+                    callback = detected_callback[0]
+                    if callback is not None:
+                        callback()
+
+                    if new_message_callback is not None:
+                        state = "ACTIVE"
+                    continue
+
+            elif state == "ACTIVE":
```

Comment:
what if you have, e.g., 5 silences, then 100 non-silences, then 9 silences, then 100 non-silences, then 2 silence? Basically audio with hesitation. I think your code doesn't handle that?

Reply:
If the keyword is spoken, and then there is "intermittent audio" afterward, the code will keep recording until silence is heard for enough counts. Now that you bring this up, I think it is a problem. It seems like a reasonable solution would be to add a timeout to the recording. The timeout can be user definable (just like the silence threshold). What do you think? If you agree I can implement this quickly.